x-pack/plugins/apm/server/routes/services.ts

```diff
@@ -33,7 +34,10 @@ export const servicesRoute = createRoute(() => ({
       ({ agentName }) => agentName as AgentName
     );
     const apmTelemetry = createApmTelementry(agentNames);
-    storeApmServicesTelemetry(context.__LEGACY.server, apmTelemetry);
+    const savedObjectsClient = await getInternalSavedObjectsClient(core);
+    storeApmServicesTelemetry(savedObjectsClient, apmTelemetry).catch(error => {
+      context.logger.error(error.message);
+    });
```

Comment:
it was necessary to add a `catch` here since the CI test saw an unresolved promise when there was a failure to persist telemetry attributes and failed CI.

Reply:
@watson I recall that you added this check (throw on unresolved promise) to the test runner. Did you also enable it at runtime in dev mode?

Reply:
when i ran the api integration tests locally, the same unresolved promise error shows in the server log, but the test runner did not mark it as a failure

Reply:
I was just thinking whether this also would throw during normal development and not just when running tests

Reply:
Yes, the check was added in #51431.

For CI the magic is here:

https://github.com/elastic/kibana/blob/5bb24810e918c27bc0a38b02444a580332a99b8e/src/dev/ci_setup/setup_env.sh#L17

In dev-mode (when running `yarn start`), the magic is here:

https://github.com/elastic/kibana/blob/5bb24810e918c27bc0a38b02444a580332a99b8e/package.json#L56

I intended it to run during dev-mode, so that developers would not be surprised by weird errors in CI that they might as well catch early on in development.

Note that if you manually run `node scripts/kibana` it will not throw on deprecations.

Reply:
Btw, normally I would think that you should pass the error on to the UI so that the user could be notified that something went wrong instead of just logging it to STDERR. But I'm not sure what this code does and if it makes sense to do one or the other in this case. Just be sure you thought it through 🙂

Reply:
> normally I would think that you should pass the error on to the UI so that the user could be notified that something went wrong instead of just logging it to STDERR.

Agree! However, the operation in this case is a background job which is implicitly initiated by the user but the response of the API doesn't rely on it.

@ogupte I think we talked about adding a comment to make this more obvious